### PR TITLE
Do not include endpoint config in types_stub.h

### PR DESCRIPTION
**Pull request by andy31415**
Including data types should not have a sideffect of including / depending on app-specific endpoint configuration.

This removes endpoint_config/gen_config from types_stub and starded adding direct includes where these seem needed (probably not exhausinve, relying on CI to test compilation for now).

Without this change, Accessors.h is not directly usable without depending on a codegen bit because of a dependency of `Accessors.h -> af-types.h -> types_stub.h -> endpoint_config.h`

`af-types.h` seemed to want some of gen/endpoint config flags (specifically to define structures depending on what clusters are selected), changed this like:
    - removed some unused clusters
    - moved scenese table entry to scenes.h

### CODE REVIEW

File: src/app/clusters/application-basic-server/application-basic-server.cpp [hunk 1]
```diff
@@ -31,8 +31,11 @@
 #endif // CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED
 #include <app/data-model/Encode.h>
 #include <app/util/attribute-storage.h>
+#include <app/util/config.h>
 #include <platform/CHIPDeviceConfig.h>
 
+#include <app/util/config.h>
+
```
Comment (bzbarsky-apple) [inline src/app/clusters/application-basic-server/application-basic-server.cpp:38]:
```suggestion
```

File: src/app/clusters/account-login-server/account-login-server.cpp [hunk 2]
```diff
@@ -28,12 +28,15 @@
...
 #include <app/util/af.h>
+#include <app/util/config.h>
 #include <platform/CHIPDeviceConfig.h>
 
 #if CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED
 #include <app/app-platform/ContentAppPlatform.h>
 #endif // CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED
 
+#include <app/util/config.h>
+
```
Comment (bzbarsky-apple) [inline src/app/clusters/account-login-server/account-login-server.cpp:39]:
```suggestion
```

File: src/app/clusters/application-launcher-server/application-launcher-server.cpp [hunk 3]
```diff
@@ -31,13 +31,16 @@
...
+#include <app/util/config.h>
 #if CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED
 #include <app/app-platform/ContentAppPlatform.h>
 #endif // CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED
 #include <app/data-model/Encode.h>
 #include <app/util/attribute-storage.h>
 #include <platform/CHIPDeviceConfig.h>
 
+#include <app/util/config.h>
+
```
Comment (bzbarsky-apple) [inline src/app/clusters/application-launcher-server/application-launcher-server.cpp:43]:
```suggestion
```

File: src/app/clusters/audio-output-server/audio-output-server.cpp [hunk 4]
```diff
@@ -31,8 +31,11 @@
 #include <app/ConcreteCommandPath.h>
 #include <app/data-model/Encode.h>
 #include <app/util/attribute-storage.h>
+#include <app/util/config.h>
 #include <platform/CHIPDeviceConfig.h>
 
+#include <app/util/config.h>
+
```
Comment (bzbarsky-apple) [inline src/app/clusters/audio-output-server/audio-output-server.cpp:38]:
```suggestion
```

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
I am 99% sure color-control-server needs to include config.h as well...  We really need that lint that checks for the defines from gen_config.h in files and make sure those files include config.h

**Review by bzbarsky-apple — APPROVED** [review]
Actually, I guess color-control-server is in this PR; I just missed it.

Approving, but we really do need the lint, as a followup.

**Review by tcarmelveilleux — APPROVED** [review]

**Pull request merged**